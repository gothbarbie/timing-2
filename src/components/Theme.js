import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
