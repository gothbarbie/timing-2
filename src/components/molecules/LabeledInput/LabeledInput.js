import React from 'react'
import styled from 'styled-components'

import Label from '../../atoms/Label'
import Input from '../../atoms/Input'

const LabeledInputStyle = styled.div`
  display: flex;
  flex-direction: column;
`

const LabelStyled = styled(Label)`
  margin-bottom: 1rem;
`

const LabeledInput = ({ name, label, onChange, type }) => (
  <LabeledInputStyle>
    <LabelStyled htmlFor={name}>{label}</LabelStyled>
    <Input name={name} onChange={onChange} type={type} />
  </LabeledInputStyle>
)

export default LabeledInput
