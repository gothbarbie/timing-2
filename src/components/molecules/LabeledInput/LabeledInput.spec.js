import React from 'react'
import { shallow } from 'enzyme'

import { theme } from '../../Theme'
import LabeledInput from './LabeledInput'

describe('generic/molecules/LabeledInput', () => {
  it('renders', () => {
    const component = shallow(
      <LabeledInput
        theme={theme}
        label="Test"
        onChange={() => jest.fn()}
        name="test"
      />
    )

    expect(component).toMatchSnapshot()
  })
})
