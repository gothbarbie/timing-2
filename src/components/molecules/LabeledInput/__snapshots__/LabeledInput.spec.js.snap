// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/molecules/LabeledInput renders 1`] = `
<styled.div>
  <Styled(styled.label)
    htmlFor="test"
  >
    Test
  </Styled(styled.label)>
  <Input
    name="test"
    onChange={[Function]}
  />
</styled.div>
`;
